src/Sprite.ts

```diff
@@ -635,77 +753,88 @@ export class Sprite extends SpriteBase {
     this._speechBubble = { text: String(text), style: "think", timeout: null };
   }
 
-  *sayAndWait(text, seconds) {
-    clearTimeout(this._speechBubble.timeout);
+  public *sayAndWait(text: string, seconds: number): Yielding<void> {
+    if (this._speechBubble?.timeout) clearTimeout(this._speechBubble.timeout);
 
+    const speechBubble: SpeechBubble = { text, style: "say", timeout: null };
     let done = false;
-    const timeout = setTimeout(() => {
-      this._speechBubble.text = "";
-      this.timeout = null;
+    const timeout = window.setTimeout(() => {
+      speechBubble.text = "";
+      speechBubble.timeout = null;
       done = true;
     }, seconds * 1000);
 
-    this._speechBubble = { text, style: "say", timeout };
+    speechBubble.timeout = timeout;
+    this._speechBubble = speechBubble;
     while (!done) yield;
   }
 
-  *thinkAndWait(text, seconds) {
-    clearTimeout(this._speechBubble.timeout);
+  public *thinkAndWait(text: string, seconds: number): Yielding<void> {
+    if (this._speechBubble?.timeout) clearTimeout(this._speechBubble.timeout);
 
+    const speechBubble: SpeechBubble = { text, style: "think", timeout: null };
     let done = false;
-    const timeout = setTimeout(() => {
-      this._speechBubble.text = "";
-      this.timeout = null;
+    const timeout = window.setTimeout(() => {
+      speechBubble.text = "";
+      speechBubble.timeout = null;
       done = true;
     }, seconds * 1000);
 
-    this._speechBubble = { text, style: "think", timeout };
+    speechBubble.timeout = timeout;
+    this._speechBubble = speechBubble;
     while (!done) yield;
   }
+
+  public static RotationStyle = Object.freeze({
+    ALL_AROUND: Symbol("ALL_AROUND"),
+    LEFT_RIGHT: Symbol("LEFT_RIGHT"),
+    DONT_ROTATE: Symbol("DONT_ROTATE"),
+  });
 }
 
-Sprite.RotationStyle = Object.freeze({
-  ALL_AROUND: Symbol("ALL_AROUND"),
-  LEFT_RIGHT: Symbol("LEFT_RIGHT"),
-  DONT_ROTATE: Symbol("DONT_ROTATE")
-});
+type StageInitialConditions = {
+  width?: number;
+  height?: number;
+} & InitialConditions;
+
+export class Stage<Vars extends object = object> extends SpriteBase<Vars> {
+  public readonly width!: number;
+  public readonly height!: number;
+  public __counter: number;
 
-export class Stage extends SpriteBase {
-  constructor(initialConditions, ...args) {
-    super(initialConditions, ...args);
+  public constructor(initialConditions: StageInitialConditions, vars: Vars) {
+    super(initialConditions, vars);
 
     // Use defineProperties to make these non-writable.
     // Changing the width and height of the stage after initialization isn't supported.
     Object.defineProperties(this, {
       width: {
         value: initialConditions.width || 480,
-        enumerable: true
+        enumerable: true,
       },
       height: {
         value: initialConditions.height || 360,
-        enumerable: true
-      }
+        enumerable: true,
+      },
     });
 
-    this.name = "Stage";
-
```

Comment:
This is just unused code, right? I don't see references to `SpriteBase.name` anywhere else in this file or sb-edit's Leopard serialization code.

Reply:
Yes, and there's no real point in having it since it's redundant with the class name.